generator/src/main/java/com/scottlogic/deg/generator/outputs/dataset_writers/JsonDataSetWriter.java

```diff
@@ -22,8 +22,13 @@
     private static final SimpleDateFormat standardDateFormat = new SimpleDateFormat("dd-MM-yyyy hh:mm:ss");
 
     @Override
-    public JsonWriter openWriter(Path directory, String filenameWithoutExtension, ProfileFields profileFields) {
-        return new JsonWriter(directory.resolve(filenameWithoutExtension + ".json"), profileFields);
+    public JsonWriter openWriter(Path directory, String fileName, ProfileFields profileFields) {
+        return new JsonWriter(directory.resolve(fileName), profileFields);
+    }
+
+    @Override
+    public String getFileName(String fileNameWithoutExtension) {
```

Comment:
When does getFileName get called?

Reply:
it gets called in the `outputDataset` methods of `DirectoryOutputTarget` and `TestCaseGenerationResultWriter`